src/Splitio/Domain/EventsManagerConfig.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System.Collections.Generic;
+
+namespace Splitio.Domain
+{
+    public class EventsManagerConfig
+    {
+        public Dictionary<SdkEvent, HashSet<SdkInternalEvent>> RequireAll { get; private set; }
+        public Dictionary<SdkEvent, HashSet<SdkInternalEvent>> RequireAny { get; private set; }
+        public Dictionary<SdkEvent, HashSet<SdkInternalEvent>> Prerequisites { get; private set; }
+        public Dictionary<SdkEvent, HashSet<SdkInternalEvent>> SuppressedBy { get; private set; }
+        public Dictionary<SdkEvent, int> ExecutionLimits { get; private set; }
+
+        private EventsManagerConfig(
+            Dictionary<SdkEvent, HashSet<SdkInternalEvent>> requireAll,
+            Dictionary<SdkEvent, HashSet<SdkInternalEvent>> requireAny,
+            Dictionary<SdkEvent, HashSet<SdkInternalEvent>> prerequisites,
+            Dictionary<SdkEvent, HashSet<SdkInternalEvent>> suppressedBy,
+            Dictionary<SdkEvent, int> executionLimits)
+        { 
+            RequireAll = requireAll;
+            RequireAny = requireAny;
+            Prerequisites = prerequisites;
+            SuppressedBy = suppressedBy;
+            ExecutionLimits = executionLimits;
+        }
+
+        public static EventsManagerConfig BuildEventsManagerConfig() 
+        {
+            Dictionary<SdkEvent, HashSet<SdkInternalEvent>> requireAll = new Dictionary<SdkEvent, HashSet<SdkInternalEvent>>();
+            Dictionary<SdkEvent, HashSet<SdkInternalEvent>> prerequisites = new Dictionary<SdkEvent, HashSet<SdkInternalEvent>>();
```

Comment:
`SdkEvent.SdkReady` would be a prerequisite for `SdkEvent.SdkUpdate`

Reply:
will add